src/libs/Navigation/CustomActions.js

```diff
@@ -6,40 +6,24 @@ import lodashGet from 'lodash/get';
 import linkingConfig from './linkingConfig';
 import navigationRef from './navigationRef';
 
+/**
+ * @returns {Object}
+ */
+function getActiveState() {
+    // We use our RootState as the dispatch's state is relative to the active navigator and might not contain our active screen.
+    return navigationRef.current.getRootState();
+}
+
 /**
  * Go back to the Main Drawer
  * @param {Object} navigationRef
  */
 function navigateBackToRootDrawer() {
-    let isLeavingNestedDrawerNavigator = false;
-
-    // This should take us to the first view of the modal's stack navigator
-    navigationRef.current.dispatch((state) => {
-        // If this is a nested drawer navigator then we pop the screen and
-        // prevent calling goBack() as it's default behavior is to toggle open the active drawer
-        if (state.type === 'drawer') {
-            isLeavingNestedDrawerNavigator = true;
-            return StackActions.pop();
-        }
-
-        // If there are multiple routes then we can pop back to the first route
-        if (state.routes.length > 1) {
-            return StackActions.popToTop();
-        }
-
-        // Otherwise, we are already on the last page of a modal so just do nothing here as goBack() will navigate us
-        // back to the screen we were on before we opened the modal.
-        return StackActions.pop(0);
+    const activeState = getActiveState();
+    navigationRef.current.dispatch({
+        ...StackActions.popToTop(),
+        target: activeState.key,
```

Comment:
Please add a descriptive comment about what is happening here. And also mention why does this works and when will it not work.

Reply:
Code comment added.